File: MapA7/MapA7/Model/Expressions/VariableExpression.cs
using MapA7.Model.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model.Expressions
{
    public class VariableExpression:IExpression
    {
        private String id;

        public VariableExpression(String id) { this.id = id; }

        public int Evaluate(SymbolTable symbolTable)
        {
            if (symbolTable.ContainsKey(id)) return symbolTable.Get(id);
            else throw new CustomException("Undefined variable!");
        }

        public String Tostring() { return id; }
    }
}

File: MapGUI3/src/Model/expressions/VariableExpression.java
package Model.expressions;

import Exceptions.UndefinedVariableException;
import Model.Heap;
import Model.SymbolTable;
import Exceptions.CustomException;

public class VariableExpression extends Expression {

    private String id;

    public VariableExpression(String id) {
        this.id = id;
    }

    public int evaluate(SymbolTable<String, Integer> symTable, Heap ph) {
        if (symTable.hasKey(id)) return symTable.get(id);
        else throw new UndefinedVariableException();
    }

    public String toString() {
        return id;
    }

}
